pkg/scheduler/core/equivalence/eqivalence.go

```diff
@@ -61,36 +74,80 @@ func NewCache() *Cache {
 //
 // NodeCache objects are thread safe within the context of NodeCache,
 type NodeCache struct {
-	mu    sync.RWMutex
-	cache predicateMap
+	mu                 sync.RWMutex
+	cache              predicateMap
+	generation         uint64
+	snapshotGeneration uint64
+	// predicateGenerations stores generation numbers for predicates, incremented on
+	// predicate invalidation. Created on first update. Use 0 if does not
+	// exist.
+	predicateGenerations         []uint64
+	snapshotPredicateGenerations []uint64
```

Comment:
It would be better to include comments for all last three members, to make other developer's life easier.